adagrams/game.py

```diff
@@ -1,11 +1,96 @@
+import random
+
+
+LETTER_POOL = ['A','A','A','A','A','A','A','A','A','B','B','C','C','D',
+        'D','D','D','E','E','E','E','E','E','E','E','E','E','E','E','F',
+        'F','G','G','G','H','H','I','I','I','I','I','I','I','I','I','J',
+        'K','L','L','L','L','M','M','N','N','N','N','N','N','O','O','O',
+        'O','O','O','O','O','P','P','Q','R','R','R','R','R','R','S','S',
+        'S','S','T','T','T','T','T','T','U','U','U','U','V','V','W','W',
+        'X','Y','Y','Z']
+
+
 def draw_letters():
-    pass
+    working_letter_pool = LETTER_POOL.copy()
+    hand = []
+
+    for index in range(10):
+        drawn_letter = random.choice(working_letter_pool)
+        if drawn_letter in working_letter_pool:
+            hand.append(drawn_letter)
+        working_letter_pool.remove(drawn_letter)  
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    working_letter_bank = letter_bank.copy()    
+    word = str.upper(word)
+
+    for letter in word:
+        if letter in working_letter_bank:
+            working_letter_bank.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    score = 0
+    word = str.upper(word)
+
+    one_point = ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T']
+    two_point = ['D','G']
+    three_point = ['B', 'C', 'M', 'P']
+    four_point = ['F', 'H', 'V', 'W', 'Y']
+    five_point = ['K']
+    eight_point = ['J', 'X']
+    ten_point = ['Q', 'Z']
```

Comment:
How can we improve on this design? Maybe a different data structure? Something we can iterate over while access the scores